### PR TITLE
rssi support in sim

### PR DESCRIPTION
**Pull request by pelikhan**
Support an RSSI value from a simulator.
- [x] test hardware
- [x] test sim

### CODE REVIEW

File: libs/radio/sim/radio.ts [hunk 1]
```diff
@@ -33,7 +33,17 @@ namespace pxsim.radio {
     export function readRawPacket() {
         const state = pxsim.getRadioState();
         const packet = state.datagram.recv();
-        return new RefBuffer(packet.payload.bufferData)
+        const buf = packet.payload.bufferData;
+        const n = buf.length;
+        if (!n)
+            return undefined;
+
+        const rbuf = BufferMethods.createBuffer(n + 4);
```
Comment (mmoskal) [inline libs/radio/sim/radio.ts:41]: are these 4 bytes at the end stripped somewhere?

File: libs/radio/radio.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ namespace radio {
 
     function handleDataReceived() {
         let buffer: Buffer = readRawPacket();
-        while (buffer && buffer.length) {
+        while (buffer) {
```
Comment (pelikhan) [inline libs/radio/radio.ts:65]: readRawPacket returns null/undefined when no packet is available

File: libs/radio/sim/radio.ts (hunk 1 above)
Comment (pelikhan) [reply libs/radio/sim/radio.ts:41]: they stay around, the typescript parser knows about those

Comment (pelikhan) [reply libs/radio/sim/radio.ts:41]: https://github.com/microsoft/pxt-common-packages/blob/master/libs/radio/radio.ts#L182

Comment (pelikhan) [reply libs/radio/sim/radio.ts:41]: We need to store the RSSI per packet; not globally otherwise it is incorrect.

**Comment by pelikhan** [comment]
I did add a few checks for handling nulls.

File: libs/radio/radio.ts [hunk 3]
```diff
@@ -164,6 +163,7 @@ namespace radio {
 
     export class RadioPacket {
         public static getPacket(data: Buffer) {
+            if (!data) return undefined;
```
Comment (pelikhan) [inline libs/radio/radio.ts:166]: null packet may be fed when calling readRawPaket

**Pull request merged**